### DISCUSSION TITLE
Partman on Cluster (Master with Slaves)

### DISCUSSION THREAD

**Discussion by psyskeptic** [pgpartman/pg_partman#566] [Q&A] [answered]
Hi everyone!

I cant find any information on how to configure partman on postgres cluster.
I have cluster of 1 master and 2 slave nodes. Looks like background worker will start run_maintenance() in slave nodes two, as the all share same configuration.

How to configure it properly? I think slave shoud not do any maintenance, and just sync all changes from master.
Should i disable BGW on slave nodes somehow, or do anything else?
What if slave BGW will create new partition faster than BGW master on master, sync will be broken?

**Comment by keithf4** [comment, marked as answer]
Replicas cannot write to the database, so it's not possible for them to create child tables before the master. All newly created tables on the primary will automatically replicate. So just do the same config on all nodes in the cluster and you should be fine.

But, you know, it looks like I never did add replica detection to the maintenance function. Can't believe I've missed that all this time. All this means is that if maintenance runs on the replica before the primary, it'll throw an error that it can't write. I can code it to fail gracefully if it's run on a replica, though. Thanks for bringing this up!

**Reply by psyskeptic** [reply]
This will be helpful, otherwise logs will be polluted with false positive errors. Thanks for Answer!

**Reply by keithf4** [reply]
So it's been a while since I've worked on the background worker. I forgot that it only actually runs on whichever is the primary. During a failover, it will stop running on the previous primary and start running on the new one.

However, I think I will still add some code for gracefully calling maintenance on any node in the replication cluster since it's also still possible to not use the BGW and call maintenance manually. This lets the same cronjob (or whatever manual process) be the same on all nodes.

**Reply by keithf4** [reply]
Created https://github.com/pgpartman/pg_partman/issues/569
